MoreLinq.Test/CartesianTest.cs

```diff
@@ -24,33 +24,39 @@ public void TestCartesianIsLazy()
         /// Verify applying Cartesian to a <c>null</c> sequence results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestCartesianSequenceANullException()
         {
-            const IEnumerable<int> sequence = null;
-            sequence.Cartesian(Enumerable.Repeat(1, 10), (a, b) => a + b);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const IEnumerable<int> sequence = null;
+                sequence.Cartesian(Enumerable.Repeat(1, 10), (a, b) => a + b);
+            });
         }
 
         /// <summary>
         /// Verify passing a <c>null</c> second sequence to Cartesian results in an exception
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestCartesianSequenceBNullException()
         {
-            var sequence = Enumerable.Repeat(1,10);
-            sequence.Cartesian<int,int,int>(null, (a, b) => a + b);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var sequence = Enumerable.Repeat(1, 10);
```

Comment:
This line should be moved outside of the lambda to `Assert.Throws` as we want to only test that `Cartesian` throws.